MiniApp/Classes/core/MiniAppClient.swift

```diff
@@ -86,18 +86,28 @@ internal class MiniAppClient: NSObject, URLSessionDownloadDelegate {
         self.session.startDownloadTask(downloadUrl: url)
     }
 
-    func requestFromServer(urlRequest: URLRequest, completionHandler: @escaping (Result<ResponseData, Error>) -> Void) {
+    func requestFromServer(urlRequest: URLRequest, retry500: Int = 0, completionHandler: @escaping (Result<ResponseData, Error>) -> Void) {
         return session.startDataTask(with: urlRequest) { (result) in
             switch result {
             case .success(let responseData):
                 let statusCode = responseData.httpResponse.statusCode
-                MiniAppLogger.d("[\(statusCode)] urlRequest \(urlRequest.url?.absoluteString ?? "-") : \n\(String(data: responseData.data, encoding: .utf8) ?? "Empty response")", "🟢")
+                let logIcon = statusCode < 300 ? "🟢" : "🟠"
+                MiniAppLogger.d("[\(statusCode)] urlRequest \(urlRequest.url?.absoluteString ?? "-") : \n\(String(data: responseData.data, encoding: .utf8) ?? "Empty response")", logIcon)
 
                 if !(200...299).contains(statusCode) {
-                    return completionHandler(.failure(
-                        self.handleHttpResponse(responseData: responseData.data,
-                                                httpResponse: responseData.httpResponse)
-                        ))
+                    let failure = self.handleHttpResponse(responseData: responseData.data, httpResponse: responseData.httpResponse)
+                    if statusCode >= 500, retry500 < 5 {
+                        let backOff = 2.0
+                        let retry = retry500 + 1
+                        let waitTime = 0.5*pow(backOff, Double(retry500))
```

Comment:
attempts will be as follow:
- 0.5x2⁰ = 0.5
- 0.5x2 = 1
- 0.5x2² = 2
- 0.5x2³ = 4
- 0.5x2⁴ = 8